Moved keydown event listener off the window

AustinLeeGordon opened the pull request:
Moved keydown event listener off the window and onto the image gallery container. Prevents slides from changing while focused outside of the slider and using the left and right arrow keys. Also tested to make sure `esc` still works when in fullscreen.

natterstefan commented:
👍🏻👍🏻 would perfectly fit into the “multiple galleries on one site” topic.

xiaolin commented:
I'm on the fence about this as some user may like/need the global keydown. Maybe we can add a prop for `useWindowKeyDown` and set `false` as default?

AustinLeeGordon commented:
@xiaolin I wasn't expecting it when I experienced it but I suppose some might. Sounds like a plan to me! Want me to update the PR?

xiaolin commented:
Sure, to make it backwards compatible can we actually set `useWindowKeyDown` to `true`. This way if any one is relying on it it will still work as expected. 

Also reminder to update docs if you have time, if not I can do it 😄

AustinLeeGordon commented:
@xiaolin Yeah, I was just thinking that it would be a major version change! I'll set it to `true` by default and update the docs. Thanks!

xiaolin commented:
Was `useWindowKeyDown` ever added? I don't see it in the changes.

AustinLeeGordon commented:
@xiaolin Got busy with some other things and forgot about this! Just pushed an update with `useWindowKeyDown` in effect. I added a toggle for it in the example site for testing and added documentation. Let me know if you see anything that needs to be updated!

xiaolin commented on src/ImageGallery.jsx:
```diff
@@ -301,7 +318,8 @@ export default class ImageGallery extends React.Component {
   componentWillUnmount() {
-    window.removeEventListener('keydown', this.handleKeyDown);
+    const { useWindowKeyDown } = this.props
+    this.imageGallery.current.removeEventListener('keydown', this.handleKeyDown);
```
This looks like its already handled below on line 336.

AustinLeeGordon replied:
@xiaolin Fixed! Let me know if you need anything else!

xiaolin commented:
Everything looks good, I think the package-lock.json was removed by mistake. If you can add that back I can merge.

AustinLeeGordon commented:
> Everything looks good, I think the package-lock.json was removed by mistake. If you can add that back I can merge.

My bad. I misread one of the previous comments. Added it back!

xiaolin commented:
Yes, I can merge, but there are changes to package.json that should be reverted here. @AustinLeeGordon can you revert the  package.json? There should be no changes to package.json from these changes.

AustinLeeGordon commented:
@xiaolin Gotcha! Reverted it back to how it was before this PR.
